decisions/0005-remixing-react-router.md

```diff
@@ -0,0 +1,202 @@
+
+1. In practice, we found that the _vast_ majority of the time all you needed to reference was the `state`
+2. For scenarios in which you really do need to distinguish, we are pretty sure that in all cases, you can deduce the `type` from `state`, current location (`useLocation`), next location (`useNavigation().location`), and submission info (`useNavigation().formData`).
+
+**`useTransition().submission` is flattened**
+
+Another area that changes is the `useTransition().submission` property was removed. We found that in practice folks never really needed the submission as a standalone thing, and instead always just cared about the `formMethod` or `formData`. So we flattened them onto the navigation, so `useNavigation()` will return an object of the format:
+
+```
+{
+  state: "idle" | "loading" | "submitting";
+  location: Location;
+  formMethod?: FormMethod;
+  formAction?: string;
+  formEncType?: FormEncType;
+  formData?: FormData;
+}
+```
+
+**Backwards Compatibility**
+
+We plan to remain backwards compatible here in Remix. Very likely we'll expose the `useNavigation` hook and encourage users to move to that. And then `useTransition` will remain in a deprecated state and it will cal `useNavigation` and then backfill the `type` and `submission` properties.
+
+### `<Form method="get">` is no longer a "submission"
+
+Functionally, these two bits of code are identical, with the only difference being that in the `<form>` case you let the user determine the query value.
+
+```jsx
+<a href="/search?query=matt">Search</a>
+
+<form action="/search">
+  <input name="query" value="matt" />
+  <button type="submit">Search</button>
+</form>
+```
+
+But, in Remix we were considering the latter a "submission" such that `useTransition().state === "submitting"`. In order to ensure our "navigations" reflect the browser behavior we have changed this in the router such that GET Form submissions result in `useNavigation().state === "loading"`.
+
+**Backwards Compatibility**
+
+This will be handled in the deprecated `useTransition` hook along with the backfill of `type` and `submission` properties
+
+### Form automatic replace behavior
+
+When performing POST navigations, you don't want to end up with a duplicate entry in the history stack which makes back-button routing weird when going through the same page twice. This is further complicated in browsers that hang onto the submission info and thus have to prompt you to warn you of re-submitting your data. We'll look at a few examples to demonstrate, but the intention is that the default behavior of the router should ensure that you can't get yourself into this double-history-entry situation when using `<Form method="post">` submission navigations.
```

Comment:
interesting !
may you precise what is the current/future behaviour for PATH/PUT/DELETE methods ?

Reply:
They all behave the same as `POST` when JS is enabled: https://reactrouter.com/en/dev/components/form#replace